jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -440,6 +440,8 @@
                 result.add(child);
             } else if (fragment.getNonJavaResources().length > 0) { // some package has non-java files
                 result.add(fragment);
+            } else if (!fragment.hasSubpackages()) {
+                result.add(fragment);
             }
```

Comment:
looks like you could add the child directly `result.add(child)`. No need of check the conditions.

Reply:
That will make every sub packages get rendered in explorer (including the default package) but without any content. For all the empty packages, just show the leaf ones looks much better.

<img width="545" alt="Screen Shot 2020-08-14 at 10 57 59 AM" src="https://user-images.githubusercontent.com/6193897/90209096-09ea7680-de1d-11ea-9332-99c7bafc0489.png">